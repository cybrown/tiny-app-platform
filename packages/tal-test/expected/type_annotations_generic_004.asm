main()
  entry:
    FunctionRef       name: box_0
    DeclareLocal      name: box, mutable: false, hasInitialValue: true
    Pop               inBlock: false
    Literal           42
    Literal           1
    MakeArray
    Literal           0
    MakeRecord
    Local             box
    Call
    Pop               inBlock: false
    Literal           42
    Literal           1
    MakeArray
    Literal           0
    MakeRecord
    Local             box
    Call
box_0(item)
  entry:
    Literal           "value"
    Local             item
    Literal           1
    MakeRecord
    MakeArrayForBlock count: 1
