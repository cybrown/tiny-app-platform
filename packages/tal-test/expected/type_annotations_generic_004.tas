fun box(item: T): {value: T} { {value: item} }

box(42)
box(42)
