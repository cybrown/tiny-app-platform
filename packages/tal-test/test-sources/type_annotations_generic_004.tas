fun box<T>(item: T): {value: T} {
    {value: item}
}

box<T: number, A: string>(42)
box<T: number>(42)
