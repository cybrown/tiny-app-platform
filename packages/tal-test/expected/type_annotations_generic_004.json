[
  {
    "kind": "DeclareLocal",
    "mutable": false,
    "name": "box",
    "value": {
      "kind": "Function",
      "body": {
        "kind": "Block",
        "children": [
          {
            "kind": "Record",
            "entries": [
              {
                "kind": "RecordEntry",
                "key": "value",
                "value": {
                  "kind": "Local",
                  "name": "item"
                }
              }
            ]
          }
        ]
      },
      "parameters": [
        {
          "name": "item",
          "type": {
            "kind": "named",
            "name": "T"
          }
        }
      ],
      "returnType": {
        "kind": "record",
        "fields": {
          "value": {
            "kind": "named",
            "name": "T"
          }
        }
      },
      "genericParameters": [
        {
          "name": "T"
        }
      ]
    }
  },
  {
    "kind": "Call",
    "args": [
      {
        "kind": "PositionalArgument",
        "value": 42
      }
    ],
    "typeArgs": {
      "T": {
        "kind": "named",
        "name": "number"
      },
      "A": {
        "kind": "named",
        "name": "string"
      }
    },
    "value": {
      "kind": "Local",
      "name": "box"
    }
  },
  {
    "kind": "Call",
    "args": [
      {
        "kind": "PositionalArgument",
        "value": 42
      }
    ],
    "typeArgs": {
      "T": {
        "kind": "named",
        "name": "number"
      }
    },
    "value": {
      "kind": "Local",
      "name": "box"
    }
  }
]